File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> Majority = new HashMap<>();
        int n = nums.length / 2;
        for (int i = 0; i < nums.length; i++) {
            int Element = nums[i];
            Majority.put(Element, Majority.getOrDefault(Element, 0) + 1);
        }
        int ans = 0;
        for (int major : Majority.keySet()) {
            if (Majority.get(major) > n) {
                ans = major;
                break;
            }
        }
        return ans;
        
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=len(nums)
        ans=-1
        dici={}
        for key in nums:
            if key not in dici:
                dici[key]=1
            else:
                dici[key]=dici[key]+1
        temp=n//2
        for i in dici:
            val=dici[i]
            if val>temp:
                ans=i
        return ans